torch/utils/_config_module.py

```diff
@@ -161,6 +161,11 @@ def visit(
                 or (hasattr(value, "__module__") and value.__module__ == "typing")
                 # Handle from torch.utils._config_module import Config
                 or (isinstance(value, type) and issubclass(value, _Config))
+                # Handle from torch.serialization import LoadEndianness
+                or (
+                    hasattr(value, "__module__")
+                    and value.__module__ == "torch.serialization"
+                )
```

Comment:
cc @ezyang is it okay to do something like this?

tldr: this visitor will take any instance of type within the file as a sub config and check that its __module__ is the config module

In my case I [import an enum from `torch.serialization`](https://github.com/pytorch/pytorch/pull/143324/files#diff-284f52965247307a704d9a8ea0de055ccb76ae521a90f1c1deb5a98c5989077dR4) that is the type that one of the configs should be, but is not intended to be a subconfig.


This enum goes into the `elif isinstance(value, type):` case, but it's not intended for this to be parsed as a subconfig. Is it okay to prevent this by hardcode it like what was done for `Config` above?

Reply:
If you want to bodge it this way you can but we should figure out a better way that doesn't involve everyone adding their own special cases here. @c00w

Reply:
It feels like changing the variable name in the import to start with double underscore would make it being ignored per the first rule above? `__LoadEndianess`

Reply:
@albanD I did try this, using `__LoadEndianness` seems to cause some issues

Traceback (most recent call last):
  File "/data/users/mg1998/pytorch/test/test_serialization.py", line 26, in <module>
    from torch.utils.serialization import config as serialization_config
  File "/data/users/mg1998/pytorch/torch/utils/serialization/__init__.py", line 1, in <module>
    from . import config
  File "/data/users/mg1998/pytorch/torch/utils/serialization/config.py", line 8, in <module>
    class load:
  File "/data/users/mg1998/pytorch/torch/utils/serialization/config.py", line 10, in load
    endianness: _Optional[__LoadEndianess] = None
NameError: name '_load__LoadEndianess' is not defined. Did you mean: '__LoadEndianess'?

Reply:
I think here we can fix this by making this conditional on typechecking. But we probably should rewrite this code to ignore import statements. It's not 100% clear to me how to do that, but I'm sure it's possible.

Reply:
The usual way to filter import statement would actually be to check if the `__module__` field does not match the name of the module you're current config-ifying. It is gameable by changing this field, but at least it will be correct for everything defined in the file